fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/sorting/RecordQuerySortPlan.java

```diff
@@ -0,0 +1,219 @@
+        }
+        return sorted.map(FDBQueriedRecord::stored).map(QueryResult::of);
+    }
+
+    @Override
+    @Nonnull
+    public RecordQueryPlan getChild() {
+        return inner.getRangesOverPlan();
+    }
+
+    @Nonnull
+    public RecordQuerySortKey getKey() {
+        return key;
+    }
+
+    @Override
+    public boolean isReverse() {
+        return key.isReverse();
+    }
+
+    @Nonnull
+    @Override
+    @API(API.Status.EXPERIMENTAL)
+    public List<? extends Quantifier> getQuantifiers() {
+        return ImmutableList.of(inner);
+    }
+
+    @Nonnull
+    @Override
+    public List<? extends Value> getResultValues() {
+        return ImmutableList.of(new QueriedValue());
+    }
+
+    @Nonnull
+    @Override
+    public AvailableFields getAvailableFields() {
+        return AvailableFields.ALL_FIELDS;
+    }
+
+    @Override
+    public String toString() {
+        return getChild() + " ORDER BY " + getKey();
+    }
+
+    @Nonnull
+    @Override
+    public Set<CorrelationIdentifier> getCorrelatedToWithoutChildren() {
+        return ImmutableSet.of();
+    }
+
+    @Nonnull
+    @Override
+    public RecordQuerySortPlan rebaseWithRebasedQuantifiers(@Nonnull final AliasMap translationMap,
+                                                            @Nonnull final List<Quantifier> rebasedQuantifiers) {
+        return new RecordQuerySortPlan((Quantifier.Physical)Iterables.getOnlyElement(rebasedQuantifiers), key);
+    }
+
+    @Nonnull
+    @Override
+    public RecordQueryPlanWithChild withChild(@Nonnull final RecordQueryPlan child) {
+        return new RecordQuerySortPlan(child, key);
+    }
+
+    @Override
+    public boolean equalsWithoutChildren(@Nonnull RelationalExpression otherExpression,
+                                         @Nonnull final AliasMap equivalencesMap) {
+        if (this == otherExpression) {
+            return true;
+        }
+        if (getClass() != otherExpression.getClass()) {
+            return false;
+        }
+        final RecordQuerySortPlan other = (RecordQuerySortPlan) otherExpression;
+        return key.equals(other.key);
+    }
+
+    @SuppressWarnings("EqualsWhichDoesntCheckParameterClass")
+    @Override
+    public boolean equals(final Object other) {
+        return structuralEquals(other);
+    }
+
+    @Override
+    public int hashCode() {
+        return structuralHashCode();
+    }
+
+    @Override
+    public int hashCodeWithoutChildren() {
+        return Objects.hash(key);
+    }
+
+    @Override
+    public int planHash(@Nonnull final PlanHashKind hashKind) {
+        return PlanHashable.objectsPlanHash(hashKind, BASE_HASH, getChild(), getKey());
+    }
+
+    @Override
+    public void logPlanStructure(StoreTimer timer) {
+        timer.increment(FDBStoreTimer.Counts.PLAN_SORT);
+        getChild().logPlanStructure(timer);
+    }
+
+    @Override
+    public int getComplexity() {
```

Comment:
I made it have the same complexity and added a TODO comment, since it did not seem like backing into the design of a more robust costing model was within the intent of this branch.